### PR TITLE
infoschema, metrics: add TableByName API duration metrics for infoschema v2

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #50959 

Problem Summary:

### What changed and how does it work?

The cost of TableByName() for v2 is higher than v1, especially when the schema cache miss.
I'd like to add more metrics for them so we know what's happening

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
/test unit-test

Failed flacky test https://github.com/pingcap/tidb/issues/54379

File: pkg/metrics/grafana/tidb.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
       "type": "grafana",
       "id": "grafana",
       "name": "Grafana",
-      "version": "7.5.11"
+      "version": "7.5.17"
```
Comment (D3Hunter) [inline pkg/metrics/grafana/tidb.json:23]: can we revert those unrelated change, only copy you own panel?

Comment (D3Hunter) [reply pkg/metrics/grafana/tidb.json:23]: see this for an example https://github.com/pingcap/tidb/pull/54268/files#diff-e1c40eabcea75f285db03b26d672c728db9d6e797edc0d74f7e56af87066bf9aR13905

Comment (tiancaiamao) [reply pkg/metrics/grafana/tidb.json]: Done

**Review by tangenta — APPROVED** [review]
